app/bundles/LeadBundle/Form/Type/ContactChannelsType.php

```diff
@@ -0,0 +1,213 @@
+                        ]
+                    );
+                    $type = 'datetime';
+                } else {
+                    $attributes = array_merge(
+                        $attr,
+                        [
+                            'class' => 'form-control',
+                        ]
+                    );
+                    $type = 'date';
+                }
+                if (!$options['public_view'] || $showContactPauseDates) {
+                    $builder->add(
+                        'contact_pause_start_date_'.$channel,
+                        $type,
+                        [
+                            'widget'     => 'single_text',
+                            'label'      => false, //'mautic.lead.frequency.contact.start.date',
+                            'label_attr' => ['class' => 'text-muted fw-n label3'],
+                            'attr'       => $attributes,
+                            'format'     => 'yyyy-MM-dd',
+                            'required'   => false,
+                        ]
+                    );
+                    $builder->add(
+                        'contact_pause_end_date_'.$channel,
+                        $type,
+                        [
+                            'widget'     => 'single_text',
+                            'label'      => 'mautic.lead.frequency.contact.end.date',
+                            'label_attr' => ['class' => 'frequency-label text-muted fw-n label4'],
+                            'attr'       => $attributes,
+                            'format'     => 'yyyy-MM-dd',
+                            'required'   => false,
+                        ]
+                    );
+                }
+            }
+        }
+
+        if (isset($options['save_button']) && $options['save_button'] === true) {
+            $builder->add(
+                'ids',
+                'hidden'
+            );
+
+            $builder->add(
+                'buttons',
+                'form_buttons',
+                [
+                    'apply_text'     => false,
+                    'save_text'      => 'mautic.core.form.save',
+                    'cancel_onclick' => 'javascript:void(0);',
+                    'cancel_attr'    => [
+                        'data-dismiss' => 'modal',
+                    ],
+                ]
+            );
+        }
+
+        if (!empty($options['action'])) {
+            $builder->setAction($options['action']);
+        }
+    }
+
+    /**
+     * @see AbstractType::configureOptions()
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setRequired(['channels']);
+        $resolver->setDefaults(
+            [
+                'public_view' => false,
+                'save_button' => false,
+            ]
+        );
+    }
+
+    /**
+     * @see AbstractType::getName()
+     * {@inheritdoc}
+     */
+    public function getName()
```

Comment:
`getName` is deprecated in Symfony 2.8. and should not be used anymore. We need to use FQCN: https://github.com/symfony/symfony/blob/2.8/UPGRADE-2.8.md#form.
Same is for `alias` of form in config. Should be replaced by DI.

Reply:
Will update.

Reply:
Updated